gsoc/proposal-Category-Class-Testing.md

```diff
@@ -0,0 +1,89 @@
+
+**GitHub handle:** @ishoshani
+
+**Slack handle:** @isho
+
+**Email:** idoshoshani@gmail.com
+
+**Other Contact:**  
+
+**Website:** www.idothoughts.com
+
+### Project Details
+
+**Project Title:**  
+
+ML Category Class Testing Tool
+
+**Project Abstract/Summary (<20 words):**  
+
+A tool to allow developers to quickly test the usefulness of Category Class based features in the Web Monitoring project. 
+
+**Describe the need your project fills:**  
+
+A big issue with working towards the prioritization Algorithm in the website monitor is that ML generally does not take text input. Some process must take the string diffs and assign some number to them so we can use them as part of a data point. One suggestion was to sort them into [class categories](https://github.com/edgi-govdata-archiving/web-monitoring-processing/issues/28), and use the frequency of these being added and deleted. However, what words these categories should include and how they should be scored is up in the air. My proposal, while not in itself set out to address this problem, does set out to make this problem much faster and easier to address. 
+
+**Describe how your project meets this need:**  
+  
+  My proposal will help create an environment where once the training and testing data becomes available, any contributer can easily try to see how changes to a category class will affect the overall algorithm. By creating a format that is easy to write and test, more contributors from different areas of expertise can help in the continuing progress of the web monitoring project. 
+
+**Milestones/Timeline:**  
+
+May 15th: Make sure environment is up to date, get to know mentors, general community bonding
+
+May 30th: Begin work on Phase 1: Positive/Negative Category Testing
+
+June 20th: Finish work on phase 1, check with mentor for optimization/testing
+
+June 26th: Deliver phase 1, Begin work on graduated Category Classes
+
+July 20th: Finish working on Phase 1, check with mentor for optimization/testing
+
+July 28th: Deliver phase 2.
+
+**Deliverables:**  
+
+A command line tool in the Web Monitoring project that takes in a form with 1 or more categories and starting weights for them, and returns a list of weights based on how much each category affected the likelyhood of prediciting the page had changed. Based on mentor recommendations, this form could be either input as a CSV file or an XML file. For phase one, this will be a category of words will be assigned a positive or negative 1 to denote whether they are significant at all based on how frequently they appear in the diff of the page. For the second deliverable the tool will assign a weight to each category to grade how much it affected the decision. Alongside these code deliverables, the mentor will also get weekly progress reports detailing how the code is doing along with research reports focusing on how to assign priorities to the inputs of a ML algorithm.
```

Comment:
Make sure to run your proposal through a spell checker before you submit, to catch misspellings like "likelyhood of prediciting".

(And read through carefully to look for sentence structure issues like "For phase one, this will be a category **of words will be** assigned a positive or negative 1 to denote...".)